tracing-attributes/tests/async_fn.rs

```diff
@@ -30,6 +30,12 @@ async fn test_ret_impl_trait_err(n: i32) -> Result<impl Iterator<Item = i32>, &'
 #[instrument]
 async fn test_async_fn_empty() {}
 
+#[instrument]
+async fn repro_async_2294() {
+    #![allow(unused_variables)]
+    let i = 42;
+}
```

Comment:
should we maybe add an outer `#[deny(unused_variables)]` attribute in the outer scope, so that we also get a hard error if `#[instrument]` just swallowed the inner attribute completely? not a blocker, just a thought.